agent/agent.go

```diff
@@ -43,6 +43,11 @@ const (
 	ProtocolReconnectingPTY = "reconnecting-pty"
 	ProtocolSSH             = "ssh"
 	ProtocolDial            = "dial"
+
+	// CoderSessionErrorCode indicates that something went wrong with the session, rather than the
+	// command just returning a nonzero exit code, and is chosen as an arbitrary, high number
+	// unlikely to shadow other exit codes, which are typically 1, 2, 3, etc.
+	CoderSessionErrorCode = 229
```

Comment:
Suggestion: We could unexport this until it's needed somewhere else.

Reply:
This numeric constant really is part of the public contract of this package, in that it is the exit code you'll get if agent hits some kind of internal error with an SSH session.